### PR TITLE
[flink]Added the function of generating labels when Flink Sink is completed …

### PR DESCRIPTION
**Pull request by siyangzeng**
### Purpose

We need to enhance paimon's capabilities in flink batch processing scenarios, such as generating tags after flink batch processing is completed. This function has a very good practical effect when combined with changelog internally.

### CODE REVIEW

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/FlinkConnectorOptions.java [hunk 1]
```diff
@@ -301,6 +301,13 @@ public class FlinkConnectorOptions {
                     .withDescription(
                             "Sink committer memory to control heap memory of global committer.");
 
+    public static final ConfigOption<Boolean> SINK_FINISH_GENERATAR_TAG =
+            ConfigOptions.key("sink.finish.generatar-tag")
```
Comment (JingsongLi) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/FlinkConnectorOptions.java:305]: Can this feature be a part of `tag.automatic-creation`? We can introduce `BATCH` TagCreationMode.
The downside of this is that it cannot be orthogonal to the automatic creation of tags in streaming writing.

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/sink/SinkFinishGeneratorTagOperator.java [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Commit {@link Committable} for snapshot using the {@link CommitterOperator}.
+ * When the task is completed, the corresponding tag is generated.
+ */
+public class SinkFinishGeneratorTagOperator<CommitT, GlobalCommitT> implements
```
Comment (JingsongLi) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/sink/SinkFinishGeneratorTagOperator.java:47]: Can you create a ITCase for this feature?

Comment (JingsongLi) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/sink/SinkFinishGeneratorTagOperator.java:47]: Also ut case.

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/FlinkConnectorOptions.java (hunk 1 above)
Comment (siyangzeng) [reply paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/FlinkConnectorOptions.java:305]: Yes, we hope to use tags in batch processing scenarios and generate corresponding tags after the current task ends.

Do you mean to add a batch mode in tag.automatic-creation without adding new configuration? If that's the case, I think it's okay

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/sink/SinkFinishGeneratorTagOperator.java (hunk 2 above)
Comment (siyangzeng) [reply paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/sink/SinkFinishGeneratorTagOperator.java:47]: Okay, I'll provide it later

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/FlinkConnectorOptions.java (hunk 1 above)
Comment (JingsongLi) [reply paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/FlinkConnectorOptions.java:305]: yes, `tag.automatic-creation=batch`

Comment (siyangzeng) [reply paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/FlinkConnectorOptions.java]: Please take another look

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/sink/SinkFinishGeneratorTagOperator.java [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+/**
+ * Commit {@link Committable} for snapshot using the {@link CommitterOperator}. When the task is
+ * completed, the corresponding tag is generated.
+ */
+public class SinkFinishGeneratorTagOperator<CommitT, GlobalCommitT>
+        implements OneInputStreamOperator<CommitT, CommitT>,
+                SetupableStreamOperator,
+                BoundedOneInput {
+
+    private static final String SINK_FINISH_TAG_PREFIX = "sinkFinish-";
```
Comment (JingsongLi) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/sink/SinkFinishGeneratorTagOperator.java:53]: Only concern is here, better name?
Maybe `batch-write-2023-12-13`?

Comment (siyangzeng) [reply paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/sink/SinkFinishGeneratorTagOperator.java]: yes, updated

**Comment by JingsongLi** [comment]
Compile failed?

**Comment by JingsongLi** [comment]
+1 and merged

**Pull request closed without merging**